library/std/src/sync/mutex.rs

```diff
@@ -219,6 +219,26 @@ impl<T> Mutex<T> {
             data: UnsafeCell::new(t),
         }
     }
+
+    /// Immediately drops the guard, and consequently unlocks the mutex.
+    ///
+    /// To unlock the mutex, one can also call [`std::mem::drop`] on the guard or let the guard go
+    /// out of scope. This function exists only as a self-documenting alternative to either of
+    /// those. Note, though, that letting the guard go out of scope may result in the lock being
+    /// held longer than if you `drop` or `unlock` it.
```

Comment:
This makes it sound like the lock might be held even after of the end of the scope.

Reply:
I rewrote that part of the doc comment. What do you think?